x-pack/plugins/security_solution/server/lib/telemetry/receiver.ts

```diff
@@ -0,0 +1,292 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  Logger,
+  CoreStart,
+  ElasticsearchClient,
+  SavedObjectsClientContract,
+} from 'src/core/server';
+import { SearchRequest } from '@elastic/elasticsearch/api/types';
+import { getTrustedAppsList } from '../../endpoint/routes/trusted_apps/service';
+import { AgentService, AgentPolicyServiceInterface } from '../../../../fleet/server';
+import { ExceptionListClient } from '../../../../lists/server';
+import { EndpointAppContextService } from '../../endpoint/endpoint_app_context_services';
+import { exceptionListItemToEndpointEntry } from './helpers';
+import { TelemetryEvent, ESLicense, ESClusterInfo, GetEndpointListResponse } from './types';
+
+export class TelemetryReceiver {
```

Comment:
Yeah it's a little confusing. I think I'm comfortable renaming in the PR we discussed as a follow-on (Telemetry Orchestrator). In my first set of commits I was definitely perceiving the behavior in the second example as primary, and by lines/functions I think the majority of the complexity in that module is the "pull from ES" functionality. Since we've decided that the Orchestrator will own the Queue, it seems that it'd serve as the receiver for now, but it we determine that there's enough functionality there, we can break out orchestrator functionality under the paradigm that you've described above, and both Tasks and the Detection Engine (streaming) approach can use the portions of the Orchestrator they need.